src/js/component.js

```diff
@@ -589,22 +593,22 @@ class Component {
 
     return iconContainer;
   }
-
   /**
    * Add a child `Component` inside the current `Component`.
    *
-   * @param {string|Component} child
+   * @template {Component} T
+   * @param {string|T} child
    *        The name or instance of a child to add.
    *
-   * @param {Object} [options={}]
+   * @param {ComponentOptions} [options={}]
    *        The key/value store of options that will get passed to children of
    *        the child.
    *
    * @param {number} [index=this.children_.length]
    *        The index to attempt to add a child into.
    *
    *
-   * @return {Component}
+   * @return {T}
```

Comment:
Renders like this in the API docs with the changes to jsdoc-typeof-plugin:
<img width="677" alt="image" src="https://github.com/user-attachments/assets/fb629f1a-3386-4f1b-ba56-7a6144e16176" />